### PR TITLE
Rename to SPMD, update file structure

### PR DESCRIPTION
**Pull request by wanchaol**
This PR:
* add a demo to do simple ddp like program with distributed tensor
* updates the overall folder to "spmd", restructure the files, so
that we can include other features, the overall structure like this:

```
spmd
│   README.md
│   requirements.txt
│
└───spmd
│   │
│   └───tensor
│       │   api.py
│       │   placement_types.py
│       │   ...
│
└───tests
    │
    └───tensor
        │   test_dist_tensor.py
        │   ...
```

### CODE REVIEW

**Review by wangkuiyi — APPROVED** [review]
Thanks @wanchaol  for canonicalize the code structure! I agree that 
- we put test cases in `/tests`, side-by-side with `/spmd`, so that the `spmd` package does NOT include test cases, and
- the package name `spmd/tenosr` is more accurate than `distributed_tensor`.

File: tests/tensor/test_dist_tensor.py [hunk 1]
```diff
@@ -7,8 +7,8 @@
 from torch.testing._internal.common_utils import (
```
Comment (mrshenli) [inline tests/tensor/test_dist_tensor.py:7]: nit: `test_dist_tensor.py` -> `test_tensor.py`?

Comment (wangkuiyi) [reply tests/tensor/test_dist_tensor.py:7]: I agree with @mrshenli  on this renaming. Is it a good idea that we merge this PR and file new PRs for further fixings? My intention is to accelerate the life-cycle of each PR. But I am totally open to other ideas.

Comment (mrshenli) [reply tests/tensor/test_dist_tensor.py:7]: yep, SGTM

**Pull request merged**